File: src/main/java/com/thebluehats/server/game/world/NoFallDamage.java
package com.thebluehats.server.game.world;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;

public class NoFallDamage implements Listener {
    @EventHandler
    public void onFall(EntityDamageEvent event) {
        event.setCancelled(event.getCause() == EntityDamageEvent.DamageCause.FALL);
    }
}

File: src/main/kotlin/com/thebluehats/server/game/world/NoFallDamage.kt
package com.thebluehats.server.game.world

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageEvent

class NoFallDamage : Listener {
    @EventHandler
    fun onFall(event: EntityDamageEvent) {
        event.isCancelled = event.cause == EntityDamageEvent.DamageCause.FALL
    }
}